### PR TITLE
Add data-driven styling support for "icon-offset"

### PR DESCRIPTION
**Pull request by lucaswoj**
Replaces https://github.com/mapbox/mapbox-gl-js/pull/3729

Related to https://github.com/mapbox/mapbox-gl-test-suite/pull/194

### CODE REVIEW

File: js/data/bucket/symbol_bucket.js [hunk 1]
```diff
@@ -295,7 +295,8 @@ class SymbolBucket {
             let shapedIcon;
             if (feature.icon) {
                 const image = icons[feature.icon];
-                shapedIcon = shapeIcon(image, layout);
+                const iconOffset = this.layers[0].getLayoutValue('icon-offset', {}, feature.properties);
```
Comment (jfirebaugh) [inline js/data/bucket/symbol_bucket.js]: Second argument to getLayoutValue should be {zoom: this.zoom}.

Comment (lucaswoj) [inline js/data/bucket/symbol_bucket.js]: Good catch. This will require an update of `mapbox-gl-test-suite`.

**Comment by lucaswoj** [comment]
ref https://github.com/mapbox/mapbox-gl-test-suite/pull/195

File: js/data/bucket/symbol_bucket.js (hunk 1 above)
Comment (lucaswoj) [inline js/data/bucket/symbol_bucket.js]: Fixed & ready for final 👀

**Pull request merged**